### PR TITLE
Add trackpad gesture PointerData types

### PR DESCRIPTION
**Pull request by moffatman**
This commit adds support to the engine core for encoding trackpad gestures for to the framework. This covers zoom, rotate, and pan gestures. The changes to send trackpad gestures from each platform will come in follow-up PRs. 

1. New types of `PointerChange`: `panZoomStart`, `panZoomUpdate`, `panZoomEnd`
2. New fields on `PointerData` to handle two-finger trackpad gesture fields (`panX`, `panY`, `scale`, `rotation`)
3. ~~Implementation of system APIs to receive trackpad gestures~~ 
   - ~~Chrome OS~~
    - ~~iPadOS~~
    - ~~Linux~~
    - ~~macOS~~
    - ~~Windows Win32~~

Not implemented for

- Vanilla Android (trackpads are not used)
- Web (largely impossible due to browser restrictions)
- Windows UWP
- Fuchsia

[Design Document](https://flutter.dev/go/trackpad-gestures)

Framework PR: https://github.com/flutter/flutter/pull/89944

Addresses https://github.com/flutter/flutter/issues/23604

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: shell/platform/windows/window_win32.cc
Comment (dkwingsmt) [inline shell/platform/windows/window_win32.cc:596]: Is it possible for `direct_manipulation_owner_` to be null? If not we should just assert. (Also many other similar cases.)

File: shell/platform/windows/window_win32.h
Comment (dkwingsmt) [inline shell/platform/windows/window_win32.h:263]: Is it possible that this value coincides with some timer that is implicitly set by others? Should we use a more arbitrary value?

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
Tried to read as much code as I can, but there are certainly logic that I missed. I found no major problems. Most of the problems are questions related to the design and event model. Also some of the names seem to remain the legacy "gesture".

File: shell/platform/windows/window_win32.h
Comment (moffatman) [reply shell/platform/windows/window_win32.h:278]: The only other timer being set is in the task runner, which uses a timer ID of 0. Any future timer which uses ID 1 without checking for previous uses will notice their problem during testing since this one fires every frame.

File: shell/platform/windows/window_win32.cc
Comment (moffatman) [reply shell/platform/windows/window_win32.cc:631]: If Direct Manipulation is not supported for some reason it would be null, so I think it should not be an assert here.

File: shell/platform/windows/direct_manipulation.cc
Comment (moffatman) [reply shell/platform/windows/direct_manipulation.cc:89]: Just noticed during testing it's very unpleasant to use without the smoothing. No other platforms provide their values with this high precision, and the problem only affects the scale value.

File: shell/platform/windows/direct_manipulation.cc
Comment (moffatman) [reply shell/platform/windows/direct_manipulation.cc:49]: DirectManipulation is based around moving a "content" within a "viewport". We are sort of faking that here since our use case is not a simple content block, after each gesture the fake content needs to be reset to its original position. I'll add some explanation in the comments.

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm
Comment (moffatman) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm:1830]: No real issues with sharing the same device ID, but all platforms will be using separate ones for clicks and pan/zoom now.

File: lib/ui/window/pointer_data_packet_converter.cc
Comment (dkwingsmt) [inline lib/ui/window/pointer_data_packet_converter.cc:350]: I understand that it makes logical sense, but in practice shouldn't pan start with (0, 0)? When will it not?

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
All good, except for minor change requests.

**Comment by dkwingsmt** [comment]
@goderbauer Would you like to take a second review on this?

File: lib/ui/window/pointer_data_packet_converter.cc
Comment (moffatman) [reply lib/ui/window/pointer_data_packet_converter.cc]: You're right, It'll always start at (0,0). I will update it to put those values there, the current way is depending on the unset pan_x and pan_y being equal to 0 in some cases.

**Review by dkwingsmt — APPROVED** [review]
LGTM, still need a second review.

File: lib/ui/pointer.dart
Comment (goderbauer) [inline lib/ui/pointer.dart:40]: nit: end these with `.` (here and below).

File: lib/ui/pointer.dart
Comment (goderbauer) [inline lib/ui/pointer.dart:294]: x -> y

**Review by goderbauer — APPROVED** [review]
LGTM after doc nits are fixed.

**Comment by dkwingsmt** [comment]
@alexmarkov It did and it's causing building breakage. Reverting at #31375. I'm wondering why it passed build though.

**Comment by aam** [comment]
> Reverting at #31375. I'm wondering why it passed build though.

It looks like flutter build_and_test_linux_unopt recipe has a problem: it copies engine's `sky_engine` into flutter's `bin/cache/pkg` folder,  with intent of having flutter framework analyzed together with current engine, but that gets immediately overridden by `bin/flutter update-packages`. From https://cirrus-ci.com/task/6329915338391552 logs:
```
cd $FRAMEWORK_PATH/flutter
rm -rf bin/cache/pkg/sky_engine
mkdir -p bin/cache/pkg/
cp -r $ENGINE_PATH/src/out/host_debug_unopt/gen/dart-pkg/sky_engine bin/cache/pkg/
bin/flutter update-packages --local-engine=host_debug_unopt
...
Downloading package sky_engine...                                  114ms
...
```

cc @godofredoc

**Pull request merged**
